packages/typescript/src/core/config.ts

```diff
@@ -0,0 +1,211 @@
+ *   return withSpan(
+ *     { name: 'add', inputs: { a, b } },
+ *     (span) => {
+ *       const result = a + b;
+ *       span.setOutputs({ result });
+ *       return result;
+ *     }
+ *   );
+ * }
+ * ```
+ */
+export function init(config: MLflowTracingConfig): void {
+  if (!config.tracking_uri) {
+    throw new Error('tracking_uri is required in configuration');
+  }
+
+  if (!config.experiment_id) {
+    throw new Error('experiment_id is required in configuration');
+  }
+
+  if (typeof config.tracking_uri !== 'string') {
+    throw new Error('tracking_uri must be a string');
+  }
+
+  if (typeof config.experiment_id !== 'string') {
+    throw new Error('experiment_id must be a string');
+  }
+
+  // Set default Databricks config path if not provided
+  if (!config.databricks_config_path) {
+    config.databricks_config_path = path.join(os.homedir(), '.databrickscfg');
+  }
+
+  if (config.tracking_uri === 'databricks' || config.tracking_uri.startsWith('databricks://')) {
+    // Parse host and token from Databricks config file for databricks URIs
+    if (!config.host || !config.token) {
+      // Determine profile name from tracking_uri
+      let profile = 'DEFAULT';
+      if (config.tracking_uri.startsWith('databricks://')) {
+        const profilePart = config.tracking_uri.slice(13); // Remove 'databricks://'
+        if (profilePart && profilePart.length > 0) {
+          profile = profilePart;
+        }
+      }
+
+      try {
+        const { host, token } = readDatabricksConfig(config.databricks_config_path, profile);
+        if (!config.host) {
+          config.host = host;
+        }
+        if (!config.token) {
+          config.token = token;
+        }
+      } catch (error) {
+        throw new Error(
+          `Failed to read Databricks configuration for profile '${profile}': ${(error as Error).message}. ` +
+            `Make sure your ${config.databricks_config_path} file exists and contains valid credentials.`
+        );
+      }
+    }
+  } else {
+    // For self-hosted MLflow tracking server, use the tracking_uri as the host
+    config.host = config.tracking_uri;
+  }
+
+  globalConfig = { ...config };
+
+  // Initialize SDK with new configuration
+  initializeSDK();
+}
+
+/**
+ * Get the current configuration. Throws an error if not configured.
+ * @returns The current MLflow tracing configuration
+ */
+export function getConfig(): MLflowTracingConfig {
+  if (!globalConfig) {
+    throw new Error(
+      'MLflow tracing is not configured. Please call init() with host and experiment_id before using tracing functions.'
+    );
+  }
+  return globalConfig;
+}
+
+/**
+ * Read Databricks configuration from .databrickscfg file
+ * @param configPath Path to the Databricks config file
+ * @param profile Profile name to read (defaults to 'DEFAULT')
+ * @returns Object containing host and token
+ */
+export function readDatabricksConfig(configPath: string, profile: string = 'DEFAULT') {
```

Comment:
q: do we expect users to call this function? Or is this public just for testing?